test/metabase/test_setup.clj

```diff
@@ -75,6 +76,37 @@
       (assert (Table (datasets/table-name->id dataset :venues))
         (format "Loading test dataset %s failed: could not find 'venues' Table!" dataset-name)))))
 
+(defn test-wrapper
+  {:expectations-options :in-context}
+  [work]
+  (try
+    (println "clearing data")
+    ;; make sure all data is deleted from our app db
+    (jdbc/with-db-transaction [conn (db/jdbc-details @db/db-connection-details)]
+      (let [tables      ["data_migrations"
+                         "activity"
+                         "revision"
+                         "setting"
+                         "core_session"
+                         "view_log"
+                         "query_queryexecution"
+                         "report_cardfavorite"
+                         "report_dashboardcard"
+                         "report_dashboardsubscription"
+                         "report_dashboard"
+                         "report_card"
+                         "metabase_foreignkey"
+                         "metabase_fieldvalues"
+                         "metabase_field"
+                         "metabase_table"
+                         "metabase_database"
+                         "core_user"]
+            mk-del-stmt (partial str "delete from ")]
+        (dorun (map (comp (partial jdbc/db-do-prepared conn) mk-del-stmt) tables))))
```

Comment:
Don't get carried away with those `comps` and `partials` :wink: . The prevailing wisdom in Clojure-land is to prefer higher-level abstractions like `for` or `doseq` where they make sense. You could just write:
```
(doseq [table tables]
  (jdbc/execute! conn [(str "delete from " table)]))
```

A bit easier to wrap your head around :scream_cat: :+1: